content/departments/product-engineering/engineering/cloud/devops/managed/v1.1/mi1-1_restore_process.md

```diff
@@ -0,0 +1,84 @@
+# Restore a managed instance v1.1 when it is completely gone
+
+<span class="badge badge-note">SOC2/CI-110</span>
+
+> Are you experiencing a zonal failure? Follow the [failover process](./mi1-1_failover_process.md)
+
+In MI v1.1, we now have resources running outside the VM, i.e Cloud SQL, hence the restoration comes with multiple stages. Perform the restoration in the following order.
+
+## Restoring Cloud SQL
+
+Use cases:
+
+- Cloud SQL data is corrupted by a broken database migration
+- Cloud SQL data is deleted
+
+### Restore from automated backup
+
+> Below process is derived from [GCP documentation](https://cloud.google.com/sql/docs/postgres/backup-recovery/restoring#gcloud)
+
+The restortion process will be performed with `gcloud`. Learn more about [why not terraform?](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#restore_backup_context).
+
+Locate the SQL instance, note the name of the instance as `SQL_INSTANCE`
+
+```sh
+gcloud sql instances list --project $PROJECT_ID
+```
+
+List all backups, note the name of the latest (or the one right before database state is corrupted) **SUCCESSFUL** backup as `SQL_BACKUP_ID`
+
+```sh
+gcloud sql backups list --instance $SQL_INSTANCE --project $PROJECT_ID
+```
+
+Restore the backup to the current instance. **IMPORTANT** This will override all current data with the backup.
+
+```sh
+gcloud sql backups restore $SQL_BACKUP_ID --restore-instance $SQL_INSTANCE --project $PROJECT_ID
+```
+
+## Restore the compute instance (VM)
+
+Use cases:
+
+- Docker daemon state is corrupted
+- Application data is deleted
+- VM is deleted
+
+### Assess what is deleted
+
+Navigate to `sourcegraph-managed-$CUSTOMER` project, and look at existing compute instance.
+
+Does the VM still exist?
+
+- No, the VM is gone.
+
+  - Does the data disk `default-$CURRENT_DEPLOYMENT-data-disk` still exists?
+    - Yes, [re-create the VM with existing data disk](##re-create-the-vm-with-existing-data-disk)
+    - No, [re-create the VM with a new data disk from disk snapshot](##re-create-the-vm-with-new-data-disk-from-disk-snapshot)
+
+- Yes, follow [operation guides](../operations.md) to troubleshoot services condition. If unable to recover running services on the VM, fallback to the recovery plan above.
+
+### Re-create the VM with existing data disk
+
+1. Open [sourcegraph/deploy-sourcegraph-managed] and check out to the `$CUSTOMER` directory
+1. Run `terraform apply` to reconcile the infrastructure to its definition in code.
+1. Follow [confirm instance health](../operations.md#confirm-instance-health)
+
+### Re-create the VM with new data disk from disk snapshot
+
+1. Go to https://console.cloud.google.com/compute/snapshots (make sure you select the right project) and find the latest snapshot, note the name
```

Comment:
We could use this command here:
```
 gcloud compute snapshots list --sort-by=~creationTimestamp
```
this sorts snapshot in desc order (by using `~`).